### ISSUE TITLE
Model Details

### ISSUE THREAD

**Issue by KennyNH** [NetManAIOps/ChatTS#29]
您好，我在阅读huggingface上ChatTS-14B的modeling_qwen2.py脚本时时遇到一些细节问题想请教一下。
问题主要是在 Line1312-1433 (Qwen2TSForCausalLM._merge_input_ids_with_time_series_features函数内)：

1. Line 1403（final_labels[batch_indices, text_to_overwrite] = labels[batch_indices, non_ts_indices]）中，为什么labels用的索引和input text的索引用的一样，都是non_ts_indices呢？labels为什么不需要重新计算自己的non_ts_indices索引？
2. Line 1423（position_ids = (final_attention_mask.cumsum(-1) - 1).masked_fill_((final_attention_mask == 0), 1)）中，为什么final_attention_mask == 0的位置要通过mask fill设置成1呢？
3. Line 1427-1431对input text embedding中的pad tokens做了masking；那么time series tokens的embeddings是否需要做masking呢？

**Comment by xiez22** [comment]
您好！我的这段代码是参考LLaVA写的：https://github.com/huggingface/transformers/blob/38f9f10dd9240619ea17fb6c7acb51b3bc592232/src/transformers/models/llava/modeling_llava.py#L297

1. labels和input text的位置应该是一一对应的（本质上是做next token prediction，所以labels和input text其实是一样的），因此二者索引理论上是一样的。只不过在最后计算loss的时候，你可以看到模型做了一个label shift的操作，这样就让模型输出的next token和label对应上
2. 这个也是LLaVA里面的设置，我猜测可能的原因是：将position_ids设置为0可能会出现一些bug，因此将其设置为1。事实上，由于已经被mask掉了，所以这个position_ids设置为任意值都不会影响结果
3. final_embedding里面，已经将time series embedding插入到input text embedding中了，所以不需要单独做masking

**Comment by KennyNH** [comment]
感谢回复！

**Comment by KennyNH** [comment]
感谢！
